src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp

```diff
@@ -213,18 +214,32 @@ void S3ObjectStorage::listPrefix(const std::string & path, BlobsPathToSize & chi
 void S3ObjectStorage::removeObject(const std::string & path)
 {
     auto client_ptr = client.get();
-    Aws::S3::Model::ObjectIdentifier obj;
-    obj.SetKey(path);
+    auto settings_ptr = s3_settings.get();
 
-    Aws::S3::Model::Delete delkeys;
-    delkeys.SetObjects({obj});
+    // If chunk size is 0, only use single delete request
+    // This allows us to work with GCS, which doesn't support DeleteObjects
+    if (settings_ptr->objects_chunk_size_to_delete == 0)
+    {
+        Aws::S3::Model::DeleteObjectRequest request;
+        request.SetBucket(bucket);
+        request.SetKey(path);
+        auto outcome = client_ptr->DeleteObject(request);
 
-    Aws::S3::Model::DeleteObjectsRequest request;
-    request.SetBucket(bucket);
-    request.SetDelete(delkeys);
-    auto outcome = client_ptr->DeleteObjects(request);
+        throwIfError(outcome);
+    }
+    else
```

Comment:
Let's remove this branch in case of single object?

Reply:
So the thing that stopped me from doing that originally was the comment around DeleteObject before the change in https://github.com/ClickHouse/ClickHouse/commit/c18cdec77c5 - I wasn't sure if there was some kind of edge case that always using DeleteObjects was working around in S3? Thoughts?

Reply:
I don't know anything about such limitations. Let's just try and see what will happen in tests.